mantidproject/mantid
# Error reporting: Display reporter on application segfaults

**Description of work**

This PR updates the ErrorReporter dialog so that it catches hard crashes to desktop as well as errors which reach the last chance error handler.

**To test:**

* These tests need to be done on a package version of mantid rather than a local build.
* It also needs to be tested on all OS's

* Open mantid
* Turn usage reporting on/restart mantid
* Run the segfault algorithm and check that the error report dialog appears.
* Check that no report is sent if you select no report
* Check that nothing appears if usage reporting is turned off.




Fixes #22293  . 

**Release Notes** 
Has been added to release notes.


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
